tests/unit/scribbles/test_transforms.py

```diff
@@ -0,0 +1,292 @@
+        (1, NUM_SLICES, HEIGHT, WIDTH),
+    ),
+]
+
+TEST_CASE_MAKE_ISEG_UNARY_TX = [
+    # 2D case
+    (
+        {
+            "image": "image",
+            "logits": "prob",
+            "scribbles": "scribbles",
+            "scribbles_bg_label": 2,
+            "scribbles_fg_label": 3,
+        },
+        {"image": two_dim_data["image"], "prob": two_dim_data["prob"], "scribbles": two_dim_data["prob"][[1], ...] + 2},
+        {"target": two_dim_data["prob"]},
+        (2, HEIGHT, WIDTH),
+    ),
+    # 3D case
+    (
+        {
+            "image": "image",
+            "logits": "prob",
+            "scribbles": "scribbles",
+            "scribbles_bg_label": 2,
+            "scribbles_fg_label": 3,
+        },
+        {
+            "image": three_dim_data["image"],
+            "prob": three_dim_data["prob"],
+            "scribbles": three_dim_data["prob"][[1], ...] + 2,
+        },
+        {"target": three_dim_data["prob"]},
+        (2, NUM_SLICES, HEIGHT, WIDTH),
+    ),
+]
+
+TEST_CASE_MAKE_LIKE_HIST_TX = [
+    # 2D case
+    (
+        {"image": "image", "scribbles": "scribbles", "scribbles_bg_label": 2, "scribbles_fg_label": 3},
+        {"image": two_dim_data["target"], "scribbles": two_dim_data["prob"][[1], ...] + 2},
+        {"target": two_dim_data["prob"]},
+        (2, HEIGHT, WIDTH),
+    ),
+    # 3D case
+    (
+        {"image": "image", "scribbles": "scribbles", "scribbles_bg_label": 2, "scribbles_fg_label": 3},
+        {"image": three_dim_data["target"], "scribbles": three_dim_data["prob"][[1], ...] + 2},
+        {"target": three_dim_data["prob"]},
+        (2, NUM_SLICES, HEIGHT, WIDTH),
+    ),
+]
+
+
+TEST_CASE_ADD_BG_ROI = [
+    (
+        {"scribbles": "scribbles", "roi_key": "roi", "scribbles_bg_label": 2, "scribbles_fg_label": 3},
+        {
+            "scribbles": np.zeros((1, NUM_SLICES, HEIGHT, WIDTH), dtype=np.float32),
+            "roi": [
+                NUM_SLICES // 2 - 4,
+                NUM_SLICES // 2 + 4,
+                HEIGHT // 2 - 8,
+                HEIGHT // 2 + 8,
+                WIDTH // 2 - 16,
+                WIDTH // 2 + 16,
+            ],
+        },
+        (1, NUM_SLICES, HEIGHT, WIDTH),
+    ),
+]
+
+
+class TestScribblesTransforms(unittest.TestCase):
+    @parameterized.expand(TEST_CASE_ADD_BG_ROI)
+    def test_add_bg_roi_transform(self, input_param, test_input, expected_shape):
+        result = AddBackgroundScribblesFromROId(**input_param)(test_input)
+        mask = result["scribbles"].astype(bool)
+        mask[
+            :,
+            test_input["roi"][0] : test_input["roi"][1],
+            test_input["roi"][2] : test_input["roi"][3],
+            test_input["roi"][4] : test_input["roi"][5],
+        ] = True
+        target = result["scribbles"]
+        target[~mask] = input_param["scribbles_bg_label"]
+
+        np.testing.assert_allclose(target, mask.astype(np.float32) * result["scribbles"], rtol=1e-4)
```

Comment:
isn't the target integer results? why is this comparison done in `float32` with a tolerance?

is the transform here not deterministic?

Reply:
This transform works on a "scribbles" input data, which is received from slicer as fp32 input. You are right that this could be done with integers, but it makes it easier to mix scribbles with logits and input image volume if it is fp32.

The transform, therefore, does not impose conversion of scribbles to integers or float32. It just adds values on top of what is already received (be it interger or fp32). 

Maybe there is a better way to be less ambiguous here? Perhaps two tests with 1) integers and 2) fp32 as input?

Reply:
I have updated with two points:

1. Compare output directly without conversion to any data type
       `np.testing.assert_allclose(input_data, result["scribbles"], rtol=1e-4)`
2. Run the transform once with fp32 input and again with integer input. 

Let me know if this addresses your comment?